### PR TITLE
fix/2539: exclude guest users from timesheet

### PR DESCRIPTION
**Pull request by yuhanga-rtc**
### Summary

Functions which added the new predicate:

- EmployeeManagerRepositoryImpl.java (findManagerSupervisingEmployee)
- EmployeeRepositoryImpl.java (findEmployeeByNameEmail, findEmployeeByName)
- EmployeeTeamRepositoryImpl.java (countAvailableEmployeesByTeamIdsAndDate, getEmployeesByTeamIds, getEmployeesByTeamIds)
- TeamRepositoryImpl.java (findEmployeesInManagerLeadingTeams, findEmployeesInTeamByTeamId, findEmployeesByTeamIds)

### How to test

1.

### Project Checklist

- [ ] Changes build without any errors
- [ ] Have written adequate test cases
- [ ] Done developer testing in
  - [ ] Chrome
  - [ ] Firefox
  - [ ] Safari
- [ ] Code is formatted with `npm run format`
- [ ] Code is linted with `npm run check-lint`
- [ ] No unnecessary comments left in code
- [ ] Made corresponding changes to the documentation

#### Other

- [ ] New atomic components added
- [ ] New molecules added
- [ ] New pages(routes) added
- [ ] New dependencies installed

### PR Checklist

- [ ] Pull request is raised from the correct source branch
- [ ] Pull request is raised to the correct destination branch
- [ ] Pull request is raised with correct title
- [ ] Pull request is self reviewed
- [ ] Pull request is self assigned
- [ ] Suitable pull request status labels are added (`ready-for-code-review`)

### Additional Information
-

### CODE REVIEW

**Review by shakila-kamalasena — APPROVED** [review]

**Review by shakila-kamalasena — DISMISSED** [review]

**Review by ThinuwanW — COMMENTED** [review]
## 🤖 Claude Code Review

This PR adds a LEFT-join predicate to two Criteria queries to exclude PM_GUEST_EMPLOYEE (guest) users, in line with the stated goal of excluding guests from timesheet views. The two predicate additions are technically correct — the LEFT join plus the `isNull() OR notEqual()` guard properly preserves employees that have no role or a non-guest role while filtering guests, and all metamodel/imports are in place. My concerns are about completeness and scope: one timesheet path is left unfiltered, and one of the modified queries is not actually a timesheet query.

Found **6** new issue(s): 🔴 1 important, 🟡 1 suggestion(s)

File: backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/TeamRepositoryImpl.java [hunk 1]
```diff
@@ -6,6 +6,8 @@
 import com.skapp.community.peopleplanner.model.Employee;
 import com.skapp.community.peopleplanner.model.EmployeeManager;
 import com.skapp.community.peopleplanner.model.EmployeeManager_;
+import com.skapp.community.peopleplanner.model.EmployeeRole;
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/TeamRepositoryImpl.java:9]: 🤖 **Claude** · 💬 **Note**

📁 **File-level observation**

Incomplete fix — a timesheet path still returns guests. The team time-record summary (TimeServiceImpl.getTimeRecordSummaryDetails, line ~1555) builds its employee list from TeamRepositoryImpl.findEmployeesInTeamByTeamId, which was NOT updated to exclude PM_GUEST_EMPLOYEE. As a result a guest who is a team member will still appear in the per-team timesheet summary, contradicting the PR's stated goal of excluding guests from timesheet. If guests should be excluded from all timesheet views, apply the same LEFT-join/pmRole predicate to findEmployeesInTeamByTeamId (and consider findEmployeesByTeamIds, which is similarly unfiltered). Please confirm whether these paths are in scope.

File: backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/EmployeeRepositoryImpl.java [hunk 2]
```diff
@@ -1218,6 +1218,10 @@ public List<Employee> findEmployeeByName(String keyword) {
 		Join<Employee, User> userJoin = root.join(Employee_.user);
 		predicates.add(criteriaBuilder.notEqual(userJoin.get(User_.isActive), false));
 
+		Join<Employee, EmployeeRole> roleJoin = root.join(Employee_.employeeRole, JoinType.LEFT);
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/EmployeeRepositoryImpl.java:1220]: 🤖 **Claude** · 💬 **Note**

Scope check: findEmployeeByName is not used by any timesheet code. Its only callers are PeopleServiceImpl.searchEmployeesAndTeamsByKeyword (the people/team keyword-search analytics feature). So this change silently alters the behavior of that search (guests will now disappear from name search results) while not affecting the timesheet at all. Either this is intentional cross-feature cleanup — in which case it's worth calling out in the PR description — or the guest exclusion belongs at the timesheet layer instead of this shared repository method. Please confirm the intent.

File: backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/TeamRepositoryImpl.java [hunk 3]
```diff
@@ -125,6 +128,10 @@ public Page<Employee> findEmployeesInManagerLeadingTeams(List<Long> teamIds, Pag
 		predicates.add(criteriaBuilder.equal(user.get(User_.isActive), Boolean.TRUE));
 		predicates.add(criteriaBuilder.equal(root.get(Employee_.ACCOUNT_STATUS), AccountStatus.ACTIVE));
 
+		Join<Employee, EmployeeRole> roleJoin = root.join(Employee_.employeeRole, JoinType.LEFT);
```
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/TeamRepositoryImpl.java:131]: 🤖 **Claude** · 💬 **Note**

Code duplication: the identical 3-line predicate block (LEFT join on employeeRole + `pmRole IS NULL OR pmRole != PM_GUEST_EMPLOYEE`) now appears verbatim here and in EmployeeRepositoryImpl.findEmployeeByName. Consider extracting a small shared helper (e.g. a static method in PeopleUtil that takes the CriteriaBuilder and the Employee root/roleJoin and returns the Predicate) so the guest-exclusion rule lives in one place and future timesheet queries can reuse it consistently.

File: backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/TeamRepositoryImpl.java (hunk 1 above)
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/TeamRepositoryImpl.java:9]: 🤖 **Claude** · 💬 **Note**

📁 **File-level observation**

No test coverage for the new behavior. There are currently no tests referencing findEmployeeByName, findEmployeesInManagerLeadingTeams, or PM_GUEST_EMPLOYEE. Given the fix's purpose, add at least one test asserting that a PM_GUEST_EMPLOYEE is excluded and that an employee with a null pmRole is still returned (verifying the LEFT-join branch), to guard against regressions.

File: backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/TeamRepositoryImpl.java (hunk 3 above)
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/TeamRepositoryImpl.java:131]: 🤖 **Claude** · 🔴 **Important**

Incomplete fix for a second timesheet path: getManagerAttendanceSummary (TimeServiceImpl ~line 754-765) builds its employee list by taking findEmployeesInManagerLeadingTeams (now guest-filtered by this PR) AND, when the caller is an ATTENDANCE_MANAGER, addAll(employeeManagerDao.findManagerSupervisingEmployee(userId)). findManagerSupervisingEmployee (EmployeeManagerRepositoryImpl.java:46) applies NO pmRole/PM_GUEST_EMPLOYEE filter, so any guest that sits under the manager via an EmployeeManager relationship is re-added to employeeList and its worked/break hours are counted in the attendance summary — bypassing this PR's exclusion. This is a distinct path from the getTimeRecordSummaryDetails/findEmployeesInTeamByTeamId leak already noted in the first pass. Fix: add the same 'pmRole IS NULL OR pmRole != PM_GUEST_EMPLOYEE' predicate to findManagerSupervisingEmployee (or filter guests out of employeeList after the union) so all three aggregation paths are consistent.

File: backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/EmployeeRepositoryImpl.java (hunk 2 above)
Comment (ThinuwanW) [inline backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/EmployeeRepositoryImpl.java:1220]: 🤖 **Claude** · 🟡 **Suggestion**

Semantic divergence in guest-exclusion idioms (separate from the identical-block duplication already noted). The new code here (and in TeamRepositoryImpl) uses a LEFT join plus 'pmRole IS NULL OR pmRole != PM_GUEST_EMPLOYEE', which correctly keeps employees whose pmRole is NULL. But the pre-existing sibling method findEmployeeByNameEmail in this same file (~lines 475-478) uses an INNER join plus a bare notEqual(pmRole, PM_GUEST_EMPLOYEE); in SQL 'pmRole != GUEST' is UNKNOWN/false when pmRole IS NULL, so that method silently drops every employee with a null pmRole. The two idioms now diverge for the same concept. Recommend extracting a single shared helper predicate (null-safe form) and aligning findEmployeeByNameEmail onto it — or, if its stricter behavior is intentional, documenting why they differ.

Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/EmployeeRepositoryImpl.java:1220]: This is intentional

File: backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/TeamRepositoryImpl.java (hunk 3 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/TeamRepositoryImpl.java:131]: No need

File: backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/TeamRepositoryImpl.java (hunk 1 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/TeamRepositoryImpl.java:9]: Done

Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/TeamRepositoryImpl.java:9]: Done

File: backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/TeamRepositoryImpl.java (hunk 3 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/TeamRepositoryImpl.java:131]: Done

File: backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/EmployeeRepositoryImpl.java (hunk 2 above)
Comment (yuhanga-rtc) [reply backend/src/main/java/com/skapp/community/peopleplanner/repository/impl/EmployeeRepositoryImpl.java:1220]: It is a heavy refactoring

**Review by shakila-kamalasena — DISMISSED** [review]

**Review by akilarootcode — APPROVED** [review]

**Review by akilarootcode — DISMISSED** [review]

**Review by akilarootcode — APPROVED** [review]

**Review by ThinuwanW — APPROVED** [review]
